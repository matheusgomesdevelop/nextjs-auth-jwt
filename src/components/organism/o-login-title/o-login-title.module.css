.o_loginTitle {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    text-align: center;
}

.o_loginTitle > h2 {
    margin: var(--spacing-24) var(--spacing-12);
    white-space: nowrap;
}