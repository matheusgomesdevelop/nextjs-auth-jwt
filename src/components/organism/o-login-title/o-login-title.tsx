// Atoms
import { A_Avatar, A_Title, A_Text } from "@/components/atoms";

// StyleSheet
import styles from "./o-login-title.module.css";

export interface I_OLoginTitle {
  image: string;
  title: string;
  username: string;
}

export const O_LoginTitle = ({ image, title, username }: I_OLoginTitle) => {
  return (
    <div data-testid="o-login-title" className={styles.o_loginTitle}>
      <A_Avatar
        image={image}
        label={username[0]}
        size="normal"
        shape="circle"
      />
      <A_Title variant="h1" className={styles.a_title}>
        {title}
      </A_Title>
      <A_Text variant="fwReg-fs16-gray500">{username}</A_Text>
    </div>
  );
};
