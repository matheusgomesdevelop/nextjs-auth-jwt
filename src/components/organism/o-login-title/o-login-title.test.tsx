import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

// Organism
import { O_LoginTitle } from "./o-login-title";

describe("Deve renderizar o o-login-title, corretamente", () => {
  // ========================================
  it("Deve renderizar o a-avatar", () => {});
  // ========================================

  it("Deve renderizar o a-title", () => {});
  // ========================================

  it("Deve renderizar o a-text", () => {});
  // ========================================

  it("Deve preservar a estrutura visual do componente", () => {
    const get_o_login_title = renderer
      .create(
        <O_LoginTitle
          image=""
          title="NextJS: Auth + JWT"
          username="@matheusgomesdev"
        />
      )
      .toJSON();

    expect(get_o_login_title).toMatchSnapshot();
  });
  // ========================================
});
