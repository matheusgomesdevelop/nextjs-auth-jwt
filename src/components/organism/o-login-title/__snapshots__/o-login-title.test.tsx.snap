// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o o-login-title, corretamente Deve preservar a estrutura visual do componente 1`] = `
<div
  className="o_loginTitle"
  data-testid="o-login-title"
>
  <div
    className="p-avatar p-component p-avatar-circle"
    data-pc-name="avatar"
    data-pc-section="root"
    data-testid="a-avatar"
    style={{}}
  >
    <span
      className="p-avatar-text"
      data-pc-section="label"
    >
      @
    </span>
  </div>
  <h1
    className="a_title"
    data-testid="a-title"
  >
    NextJS: Auth + JWT
  </h1>
  <p
    className="a_text__fwReg_fs16_gray500"
    data-testid="a-text"
  >
    @matheusgomesdev
  </p>
</div>
`;
